### PR TITLE
AnalyzerWavefromTest Reference compare. 

### PR DESCRIPTION
**Pull request by daschuer**
Just like the signal path tests a reference waveform is used for comparison.  
The Track is currently a single jump. 
Not sure if the approach is good. Let's keep as PR until we are confident. 
This avoids to clutter the git history with 100 kB binary files.  

This is on top of https://github.com/mixxxdj/mixxx/pull/11488 which can be merged at any time.

### CODE REVIEW

**Comment by Swiftb0y** [comment]
Can you rebase now that the preliminary PR has been merged? Is this ready for review?

**Comment by daschuer** [comment]
For reference the related issues: 
https://github.com/mixxxdj/mixxx/issues/11483
https://github.com/mixxxdj/mixxx/issues/11409

The reference waveform is quite long. I will check with your results if we can reduce the length to not make our git repro to big.

**Comment by daschuer** [comment]
Now the reference waveform is only 485  bytes, OK for adding to our git tree. 
It will no longer detect the INF issue of fidlib, but we have already a test for this. 
This is now just a guard against incompatible updates to our waveform format.

File: src/test/analyserwaveformtest.cpp [hunk 1]
```diff
@@ -45,6 +45,49 @@ class AnalyzerWaveformTest : public MixxxTest {
         }
     }
 
+    void assertWaveformReference(
+            ConstWaveformPointer pWaveform,
+            const QString& reference_title) {
+        pWaveform->dump();
+
+        QFile f(QDir::currentPath() + "/src/test/reference_buffers/" + reference_title);
```
Comment (Swiftb0y) [inline src/test/analyserwaveformtest.cpp:53]: deduplicate path

File: src/test/analyserwaveformtest.cpp [hunk 2]
```diff
@@ -45,6 +45,49 @@ class AnalyzerWaveformTest : public MixxxTest {
...
+        bool pass = true;
+        // If the file is not there, we will fail and write out the .actual
+        // reference file.
+        QByteArray actual = pWaveform->toByteArray();
+        QByteArray reference;
+        if (f.open(QFile::ReadOnly)) {
+            reference = f.readAll();
+        }
```
Comment (Swiftb0y) [inline src/test/analyserwaveformtest.cpp:61]: It doesn't make sense to continue if the file couldn't be opened.
```suggestion
        ASSERT_TRUE(f.open(QFile::ReadOnly));
        QByteArray reference = f.readAll();
```

File: src/test/analyserwaveformtest.cpp [hunk 3]
```diff
@@ -45,6 +45,49 @@ class AnalyzerWaveformTest : public MixxxTest {
...
+            ConstWaveformPointer pWaveform,
+            const QString& reference_title) {
+        pWaveform->dump();
+
+        QFile f(QDir::currentPath() + "/src/test/reference_buffers/" + reference_title);
+        bool pass = true;
+        // If the file is not there, we will fail and write out the .actual
+        // reference file.
+        QByteArray actual = pWaveform->toByteArray();
+        QByteArray reference;
```
Comment (Swiftb0y) [inline src/test/analyserwaveformtest.cpp:58]: Do we really want a bitwise comparision? Wouldn't it make sense to compare the waveformdata on the basis of the actual content? Not the memory or on-disk representation (which is here assumed to be the same too).

Comment (daschuer) [reply src/test/analyserwaveformtest.cpp:58]: This was the easiest way to have two independent paths for comparing the disc representation. 
With some efforts, we can also do it on the member variables, but that is something I do not want to spend

**Comment by daschuer** [comment]
Done

**Comment by daschuer** [comment]
This is IMHO read for merge to 2.3.5. It has no user impact since it is only a unittest.

**Review by Swiftb0y — APPROVED** [review]
good enough for now, though I'd really prefer to do a semantic comparision instead of the on-disk-format bytewise comparision in the future.

**Pull request merged**